src/mongocxx/uri.hpp

```diff
@@ -32,10 +32,13 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 ///
 /// Class representing a MongoDB connection string URI.
 ///
-/// @todo return const char* instead or stringview
-/// @todo harmonize with C library (options, credentials, etc...)
+/// @todo Use optionals for all option getters. Otherwise users cannot distinguish between a case of
+/// an empty option explicitly present in the URI and the option being omitted. Also change getters
+/// that return a std::string to return a string_view since a copy is not necessary.
```

Comment:
I synced up with the URI options spec: https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst

We already had some option getters, and some URI options are implied through other getters. E.g. `w` and `wTimeoutMS` are returned via `write_concern()`.

Looking at Java's API further convinced me to return optionals:
https://mongodb.github.io/mongo-java-driver/4.1/apidocs/mongodb-driver-core/com/mongodb/ConnectionString.html

I created https://jira.mongodb.org/browse/CXX-2038 and will paste link into this comment in the next commit.

Reply:
Yes, I agree with both of these future changes.